src/main/java/de/dennisguse/opentracks/settings/PreferencesUtils.java

```diff
@@ -552,30 +551,33 @@ public static int getLayoutColumnsByDefault() {
         return resources.getInteger(R.integer.stats_custom_layout_columns_default);
     }
 
-    private static List<TypedArray> getMultiTypedArray(String key) {
-        List<TypedArray> typedArrays = new ArrayList<>();
-
-        try {
-            Class<R.array> resource = R.array.class;
-            Field field;
-            int i = 0;
-
-            do {
-                field = resource.getField(key + resources.getString(R.string.stats_custom_layout_fields_default_value_separator) + i);
-                typedArrays.add(resources.obtainTypedArray(field.getInt(null)));
-                i++;
-            } while (field != null); //TODO Catch no such field exception instead of using an endless loop.
-        } catch (Exception e) {
-            e.printStackTrace();
-            Log.e(TAG, e.getMessage());
-        }
-
-        return typedArrays;
+    private static List<TypedArray> getMultiTypedArray() {
+        return Stream.of(
+                R.array.stats_custom_layout_fields_default_value_0,
+                R.array.stats_custom_layout_fields_default_value_1,
+                R.array.stats_custom_layout_fields_default_value_2,
+                R.array.stats_custom_layout_fields_default_value_3,
+                R.array.stats_custom_layout_fields_default_value_4,
+                R.array.stats_custom_layout_fields_default_value_5,
+                R.array.stats_custom_layout_fields_default_value_6,
+                R.array.stats_custom_layout_fields_default_value_7,
+                R.array.stats_custom_layout_fields_default_value_8,
+                R.array.stats_custom_layout_fields_default_value_9,
+                R.array.stats_custom_layout_fields_default_value_10,
+                R.array.stats_custom_layout_fields_default_value_11,
+                R.array.stats_custom_layout_fields_default_value_12,
+                R.array.stats_custom_layout_fields_default_value_13,
+                R.array.stats_custom_layout_fields_default_value_14,
+                R.array.stats_custom_layout_fields_default_value_15,
+                R.array.stats_custom_layout_fields_default_value_16,
+                R.array.stats_custom_layout_fields_default_value_17,
+                R.array.stats_custom_layout_fields_default_value_18
```

Comment:
That piece of code is more clear and better to read.

But, if we decide to add a new field then we'll have to remember to add it into settings.xml and inside this stream in PreferencesUtils.java

I don't think we are going to add fields very often, so it's good for me.

Reply:
PS/ I've not tested it :D

Reply:
:D